source/initial_temperature/continental_geotherm.cc

```diff
@@ -0,0 +1,257 @@
+        return (LAB_isotherm-T2)/thicknesses[2] *(depth-thicknesses[0]-thicknesses[1]) + T2;
+      // Return a constant sublithospheric temperature of 10*LAB_isotherm.
+      // This way we can combine the continental geotherm with an adiabatic profile from the input file
+      // using the "minimum" operator on the "List of initial temperature models"
+      else
+        return 10.*LAB_isotherm;
+
+    }
+
+
+
+    template <int dim>
+    void
+    ContinentalGeotherm<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.enter_subsection("Continental geotherm");
+        {
+          prm.declare_entry ("Layer thicknesses", "30000.",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of thicknesses for the bottom of the lithospheric layers,"
+                             "for a total of N+1 values, where N is the number of compositional fields."
+                             "If only one value is given, then all use the same value.  Units: $kg / m^3$");
+          prm.declare_entry ("Surface temperature", "273.15",
+                             Patterns::Double (0),
+                             "The value of the surface temperature. Units: Kelvin.");
+          prm.declare_entry ("LAB isotherm temperature", "1673.15",
+                             Patterns::Double (0),
+                             "The value of the isothermal boundary temperature assumed at the LAB "
+                             "and up to the reference depth . Units: Kelvin.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    ContinentalGeotherm<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      unsigned int n_fields = 0;
+      prm.enter_subsection ("Compositional fields");
+      {
+        n_fields = prm.get_integer ("Number of fields");
+      }
+      prm.leave_subsection();
+
+
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.enter_subsection("Continental geotherm");
+        {
+          LAB_isotherm = prm.get_double ("LAB isotherm temperature");
+          T0 = prm.get_double ("Surface temperature");
+          thicknesses = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Layer thicknesses"))),
+                                                                3,
+                                                                "Layer thicknesses");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+
+      // Retrieve the indices of the fields that represent the lithospheric layers.
+      AssertThrow(this->introspection().compositional_name_exists("upper"),ExcMessage("We need a compositional field called 'upper' representing the upper crust."));
```

Comment:
Hmm, enforcing the names to be `"upper"` and `"lower"` is really specific, but I know this was done to be consistent with the continental extension cookbook. Could we allow the name to be `upper` or `upper_crust`, etc?

Reply:
I can allow more names, but the logic for getting the composition index, e.g. ` const unsigned int id_upper = this->introspection().compositional_index_for_name("upper");`, would get much more complicated as well.

Reply:
You are right, this will get more complicated really quickly. More thoughts on this in the review summary.